benchmark/rocblas/blas1/dot.cpp

```diff
@@ -0,0 +1,145 @@
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, rocblas_handle& rb_handle, index_t size,
+         bool* success) {
+  // Google-benchmark counters are double.
+  double size_d = static_cast<double>(size);
+  state.counters["size"] = size_d;
+  state.counters["n_fl_ops"] = 2 * size_d;
+  state.counters["bytes_processed"] = 2 * size_d * sizeof(scalar_t);
+
+  // Create data
+  std::vector<scalar_t> v1 = blas_benchmark::utils::random_data<scalar_t>(size);
+  std::vector<scalar_t> v2 = blas_benchmark::utils::random_data<scalar_t>(size);
+  scalar_t res;
+
+  {
+    // Device memory allocation & H2D Copy
+    blas_benchmark::utils::HIPVector<scalar_t> d_v1(size, v1.data());
+    blas_benchmark::utils::HIPVector<scalar_t> d_v2(size, v2.data());
+    blas_benchmark::utils::HIPScalar<scalar_t> d_res(res);
+
+#ifdef BLAS_VERIFY_BENCHMARK
+    // Reference dot
+    scalar_t res_ref = reference_blas::dot(size, v1.data(), 1, v2.data(), 1);
+
+    // Rocblas verification dot
+    scalar_t res_temp = 0;
+
+    {
+      // Temp result on device (copied back to Host upon destruction)
+      blas_benchmark::utils::HIPScalar<scalar_t, true> d_res_temp(res_temp);
+      // rocBLAS function call
+      rocblas_dot_f<scalar_t>(rb_handle, size, d_v1, 1, d_v2, 1, d_res_temp);
+    }
+
+    if (!utils::almost_equal(res_temp, res_ref)) {
+      std::ostringstream err_stream;
+      err_stream << "Value mismatch: " << res_temp << "; expected " << res_ref;
+      const std::string& err_str = err_stream.str();
+      state.SkipWithError(err_str.c_str());
+      *success = false;
+    };
+#endif
+
+    auto blas_warmup = [&]() -> void {
+      rocblas_dot_f<scalar_t>(rb_handle, size, d_v1, 1, d_v2, 1, d_res);
+      CHECK_HIP_ERROR(hipStreamSynchronize(NULL));
+      return;
+    };
+
+    hipEvent_t start, stop;
+    CHECK_HIP_ERROR(hipEventCreate(&start));
+    CHECK_HIP_ERROR(hipEventCreate(&stop));
+
+    auto blas_method_def = [&]() -> std::vector<hipEvent_t> {
+      CHECK_HIP_ERROR(hipEventRecord(start, NULL));
+      rocblas_dot_f<scalar_t>(rb_handle, size, d_v1, 1, d_v2, 1, d_res);
+      CHECK_HIP_ERROR(hipEventRecord(stop, NULL));
+      CHECK_HIP_ERROR(hipEventSynchronize(stop));
+      return std::vector{start, stop};
+    };
+
+    // Warm up to avoid benchmarking data transfer
+    blas_benchmark::utils::warmup(blas_warmup);
+
+    blas_benchmark::utils::init_counters(state);
+
+    // Measure
+    for (auto _ : state) {
+      std::tuple<double, double> times =
+          blas_benchmark::utils::timef_hip(blas_method_def);
+
+      // Report
+      blas_benchmark::utils::update_counters(state, times);
+    }
+
+    blas_benchmark::utils::calc_avg_counters(state);
+  }
+};
+
+template <typename scalar_t>
+void register_benchmark(blas_benchmark::Args& args, rocblas_handle& rb_handle,
+                        bool* success) {
+  auto blas1_params = blas_benchmark::utils::get_blas1_params(args);
```

Comment:
Could you please change this to the `operator` name here e.g. `dot_params` in all the files? This would make the names consistent across all the benchmarks that we have.